fineract-provider/src/main/java/org/apache/fineract/accounting/journalentry/service/AccrualBasedAccountingProcessorForSavings.java

```diff
@@ -59,23 +59,23 @@ public void createJournalEntriesForSavings(final SavingsDTO savingsDTO) {
             if (savingsTransactionDTO.getTransactionType().isWithdrawal() && savingsTransactionDTO.isOverdraftTransaction()) {
                 boolean isPositive = amount.subtract(overdraftAmount).compareTo(BigDecimal.ZERO) > 0;
                 if (savingsTransactionDTO.isAccountTransfer()) {
-                    this.helper.createCashBasedJournalEntriesAndReversalsForSavings(office, currencyCode,
```

Comment:
I am not sure this is correct. You just changed from cash based journal entries to accrual based one without any condition. I am assuming accounting wise after this PR cash based and accrual based can be selected and both should work. But here you are overriding the behaviour. Am i missing something?